src/wp-includes/l10n/class-wp-translation-controller.php

```diff
@@ -0,0 +1,410 @@
+					unset( $this->loaded_files[ $moe->get_file() ][ $l ][ $textdomain ] );
+					return true;
+				}
+			}
+		}
+
+		return true;
+	}
+
+	/**
+	 * Unloads all translation files for a given text domain.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string $textdomain Optional. Text domain. Default 'default'.
+	 * @param string $locale     Optional. Locale. Default all locales.
+	 * @return bool True on success, false otherwise.
+	 */
+	public function unload_textdomain( string $textdomain = 'default', string $locale = null ) {
+		if ( null !== $locale ) {
+			foreach ( $this->loaded_translations[ $locale ][ $textdomain ] as $moe ) {
+				unset( $this->loaded_files[ $moe->get_file() ][ $locale ][ $textdomain ] );
+			}
+
+			unset( $this->loaded_translations[ $locale ][ $textdomain ] );
+
+			return true;
+		}
+
+		$unloaded = false;
+
+		foreach ( $this->loaded_translations as $l => $domains ) {
+			if ( ! isset( $domains[ $textdomain ] ) ) {
+				continue;
+			}
+
+			$unloaded = true;
+
+			foreach ( $domains[ $textdomain ] as $moe ) {
+				unset( $this->loaded_files[ $moe->get_file() ][ $l ][ $textdomain ] );
+			}
+
+			unset( $this->loaded_translations[ $l ][ $textdomain ] );
+		}
+
+		return $unloaded;
+	}
+
+	/**
+	 * Determines whether translations are loaded for a given text domain.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string $textdomain Optional. Text domain. Default 'default'.
+	 * @param string $locale     Optional. Locale. Default current locale.
+	 * @return bool True if there are any loaded translations, false otherwise.
+	 */
+	public function is_loaded( string $textdomain = 'default', string $locale = null ) {
+		if ( null === $locale ) {
+			$locale = $this->current_locale;
+		}
+
+		return isset( $this->loaded_translations[ $locale ][ $textdomain ] ) &&
+			array() !== $this->loaded_translations[ $locale ][ $textdomain ];
+	}
+
+	/**
+	 * Translates a singular string.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string $text       Text to translate.
+	 * @param string $context    Optional. Context for the string. Default empty string.
+	 * @param string $textdomain Optional. Text domain. Default 'default'.
+	 * @param string $locale     Optional. Locale. Default current locale.
+	 * @return string|false Translation on success, false otherwise.
+	 */
+	public function translate( string $text, string $context = '', string $textdomain = 'default', string $locale = null ) {
+		if ( '' !== $context ) {
+			$context .= "\4";
+		}
+
+		$translation = $this->locate_translation( "{$context}{$text}", $textdomain, $locale );
+
+		if ( false === $translation ) {
+			return false;
+		}
+
+		return $translation['entries'][0];
+	}
+
+	/**
+	 * Translates plurals.
+	 *
+	 * Checks both singular+plural combinations as well as just singulars,
+	 * in case the translation file does not store the plural.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param array{0: string, 1: string} $plurals    Pair of singular and plural translations.
```

Comment:
This is an array shape (which I love) but should `@type` convention be used (as well)? cf. https://wordpress.slack.com/archives/C5VCTJGH3/p1705428874962969

Reply:
Does not really work for arbitrary key-value pairs

Reply:
I think the convention in core has been to do this:

```suggestion
	 * @param array {
	 *     Pair of singular and plural translations.
	 *
	 *     @type string $0 Singular translation.
	 *     @type string $1 Plural translation.
	 * }
```

For example `wp_get_attachment_image_src()`:

https://github.com/WordPress/wordpress-develop/blob/0b7165e60d4c0d8c22c1fd736ed8a5335a8f4439/src/wp-includes/media.php#L959-L966

Reply:
Combining the two:


```suggestion
	 * @param array{0: string, 1: string} {
	 *     Pair of singular and plural translations.
	 *
	 *     @type string $0 Singular translation.
	 *     @type string $1 Plural translation.
	 * }
```

Reply:
~@westonruter Those are all for tuples, not key -> value pairs~

Edit: just had to re-read to realize this is a tuple here too 🤦  sorry!